commit 8207ea946fe96480063bfe7d177f475a21c9a525 (refs/changes/50/4250/1)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2018-06-18T11:44:27-07:00 (1 year, 4 months ago)
    
    TRITON-518 sdc-docker service crash on createVolume with missing network

diff --git a/lib/backends/sdc/volumes.js b/lib/backends/sdc/volumes.js
index 2655e80..065efc7 100644
--- a/lib/backends/sdc/volumes.js
+++ b/lib/backends/sdc/volumes.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 var assert = require('assert-plus');
@@ -126,7 +126,9 @@ function createVolume(volumeParams, options, callback) {
                 mod_networks.findNetworkOrPoolByNameOrId(volumeParams.network,
                     networkOpts,
                     function onGetNamedNetwork(getNetworkErr, network) {
-                        payload.networks = [network.uuid];
+                        if (!getNetworkErr) {
+                            payload.networks = [network.uuid];
+                        }
                         next(getNetworkErr);
                     });
             }
